csrc/ir/utils.cpp

```diff
@@ -1441,20 +1451,22 @@ bool matchMatmulPatterns(const UnaryOp* cast_op, MatmulInputs* matmul_inp) {
     }
   }
 
+  auto bias_ndims = matmul_inp->bias->as<TensorView>()->nDims();
+  auto inp_ndims = matmul_inp->mma_lhs->as<TensorView>()->nDims();
+
+  NVF_ERROR(
+      (bias_ndims == inp_ndims - 1) || (bias_ndims == inp_ndims),
+      "Bias should be 1D / 2D tensor.");
+
   // Check if bias was broadcasted
   auto* bcast = dynamic_cast<BroadcastOp*>(matmul_inp->bias->definition());
-  if (bcast != nullptr) { // mma_lhs is broadcasted
-    // Bias of shape [M, 1]
-    NVF_ERROR(
-        matmul_inp->bias->as<TensorView>()
-            ->getRootDomain()
-            .back()
-            ->isBroadcast(),
-        "Expected last dimension to be broadcasted for bias tensor.");
+  if (bcast != nullptr) {
+    // Bias of shape [M, 1] / [1, N]
+    // TODO: Allow multiple bcast axes for batched GEMMs
+    matmul_inp->bias_bcast_axis = bcast->isBroadcastDim(0) ? 0 : 1;
```

Comment:
batched GEMMs can have multiple broadcast axes in the bias. For simplicity, I have set the bcast axes for the bias to be either 0 (`bias = [1, N]`) / 1 (`bias = [M, 1]`). A more general approach would be to get the broadcastDimFlags and unsqueeze all the dims which have the flag set to `true`.

Any preference here?

Note: This change was made to support any bias shape with `addmm`. Earlier, we only supported `[M,]` or `[M,N]`.

Reply:
> A more general approach would be to get the broadcastDimFlags and unsqueeze all the dims which have the flag set to true.

How hard is this? If it's 5 lines of code as it sounds like, I'd totally do this in this PR to make code more robust. Code as is will fail quietly when batched gemms kick in.